src/query/interpreter.cpp

```diff
@@ -1399,6 +1400,141 @@ PreparedQuery PrepareDumpQuery(ParsedQuery parsed_query, std::map<std::string, T
                        RWType::R};
 }
 
+std::vector<std::vector<TypedValue>> AnalyzeGraphQueryHandler::AnalyzeGraphCreateStatistics(
+    const std::span<std::string> labels, DbAccessor *execution_db_accessor, storage::Storage::Accessor *dba) {
+  auto results = std::vector<std::vector<TypedValue>>();
+  using VertexAccessorIterable = decltype(std::declval<query::DbAccessor>().Vertices(storage::View::OLD));
+  using LPIndex = std::pair<storage::LabelId, storage::PropertyId>;
+  VertexAccessorIterable vertices = execution_db_accessor->Vertices(storage::View::OLD);
+
+  std::map<LPIndex, std::map<storage::PropertyValue, int64_t>> counter;
+
+  // initialize index
+  const auto &indices_info = execution_db_accessor->ListAllIndices();
+  std::for_each(
+      vertices.begin(), vertices.end(), [&indices_info, &counter, &labels, execution_db_accessor](const auto &vertex) {
+        std::for_each(
+            indices_info.label_property.begin(), indices_info.label_property.end(),
+            [&labels, &counter, execution_db_accessor, &vertex](const auto &info) {
+              if (const auto &has_label = vertex.HasLabel(storage::View::OLD, info.first);
+                  has_label.HasError() || !*has_label ||
+                  (labels[0] != "*" && std::find(labels.begin(), labels.end(),
+                                                 execution_db_accessor->LabelToName(info.first)) == labels.end()))
+                return;
+              if (const auto &property = vertex.GetProperty(storage::View::NEW, info.second);
```

Comment:
I think this storage::View::NEW can be kind of misleading, as if we will take into account changes which happened in this transaction, but actually we won't because we don't give possiblity to user to use this as part of transaction, and also, if there is some other transaction taking place which is changing this properties, we still won't see those changes with snapshot isolation level, and only thing that can influence this is either ANALYTICS MODE or isolation level read uncommited. On the other hand, view only has influence if we are inside same transaciton, which if I am not wrong is forbidden (and should be)